src/tools/miri/tests/pass/intrinsics-x86-sse2.rs

```diff
@@ -88,41 +72,15 @@ mod tests {
         }
         test_mm_madd_epi16();
 
-        #[target_feature(enable = "sse2")]
-        unsafe fn test_mm_mulhi_epi16() {
-            let (a, b) = (_mm_set1_epi16(1000), _mm_set1_epi16(-1001));
-            let r = _mm_mulhi_epi16(a, b);
-            assert_eq_m128i(r, _mm_set1_epi16(-16));
-        }
-        test_mm_mulhi_epi16();
-
-        #[target_feature(enable = "sse2")]
-        unsafe fn test_mm_mulhi_epu16() {
-            let (a, b) = (_mm_set1_epi16(1000), _mm_set1_epi16(1001));
-            let r = _mm_mulhi_epu16(a, b);
-            assert_eq_m128i(r, _mm_set1_epi16(15));
-        }
-        test_mm_mulhi_epu16();
-
-        #[target_feature(enable = "sse2")]
-        unsafe fn test_mm_mul_epu32() {
-            let a = _mm_setr_epi64x(1_000_000_000, 1 << 34);
-            let b = _mm_setr_epi64x(1_000_000_000, 1 << 35);
-            let r = _mm_mul_epu32(a, b);
-            let e = _mm_setr_epi64x(1_000_000_000 * 1_000_000_000, 0);
-            assert_eq_m128i(r, e);
-        }
-        test_mm_mul_epu32();
-
```

Comment:
Why remove the tests? They should still pass, right? I think we should keep them, even if they now just provide extra test coverage for the simd intrinsics. They also help ensure that we actually do support the ABI surface of core::arch.

Reply:
I can keep them if you want, but those wouldn't have landed in Miri in the first place if the functions hadn't used LLVM intrinsics from the beginning. Doesn't [miri-test-libstd](https://github.com/rust-lang/miri-test-libstd) give that extra coverage?

Reply:
We like to have basic coverage in the Miri test suite since that runs in rustc CI. 

So yeah please keep the tests. Ideally we'd invoke every public stable function at least once to make sure that codepath doesn't hit "unsupported" errors, but with the arch intrinsics that might be asking too much... but when we do have a test, there's no reason to remove it.